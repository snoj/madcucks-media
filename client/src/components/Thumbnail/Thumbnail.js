import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Thumbnail.css';

export default class Thumbnail extends Component {

    convertDate(currentDate, publishedDate) {
        publishedDate = new Date(publishedDate);
        let diff = (currentDate - publishedDate)/(1000 * 60);

        if(diff < 60) {
            return Math.floor(diff).toString(10) + " minutes ago";
        } else if(diff/60 < 2) {
            return Math.floor(diff/60).toString(10) + " hour ago";
        } else if(diff/60 < 24) {
            return Math.floor(diff/60).toString(10) + " hours ago";
        } else if(diff/(60 * 24) < 2){
            return Math.floor(diff/(60 * 24)).toString(10) + " day ago";
        } else if(diff/(60 * 24) < 30){
            return Math.floor(diff/(60 * 24)).toString(10) + " days ago";
        } else if(diff/(60 * 24 * 30) < 2){
            return Math.floor(diff/(60 * 24 * 30)).toString(10) + " month ago";
        } else if(diff/(60 * 24 * 30) < 12){
            return Math.floor(diff/(60 * 24 * 30)).toString(10) + " months ago";
        } else if(diff/(60 * 24 * 30 * 12) < 2){
            return Math.floor(diff/(60 * 24 * 30 * 12)).toString(10) + " year ago";
        } else {
            return Math.floor(diff/(60 * 24 * 30 * 12)).toString(10) + " years ago";
        }

    }

    render() {
        return(
            <div className="thumbnail-container">
                <Link to={this.props.episodeURL}>
                    <img src={this.props.imageSrc} alt=""/>
                </Link>
                <div className="thumbnail-info-container">
                    <div className="thumbnail-span-heading">{this.props.title}</div>
                    <div className="thumbnail-span-subheading">{this.props.duration} / {this.convertDate(Date.now(), this.props.date)}</div>
                </div>
            </div>
            
            
        );
    }
}
