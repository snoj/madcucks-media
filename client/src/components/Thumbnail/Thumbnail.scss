@import '../../variables.scss';

.thumbnail-container {
    
    display: flex;
    flex-direction: column;
    flex: 0 0 32%;
    background-color: $madcucks-actual-black;
    margin-top: 2em;

    @media screen and (max-width: $break-medium) {
        flex: 0 0 48%;
    }

    @media screen and (max-width: $break-small) {
        flex: 0 0 94%;
    }

    img {
        flex: 0 0 90%;
        width: 100%;
    }  

    .thumbnail-info-container {
        background-color: $madcucks-actual-black;
        flex: 0 0 10%;
        text-align: left;
        font-size: 0.8em;
        padding: 0.5em;

        .thumbnail-span-heading {
            color: white;
        }

        .thumbnail-span-subheading {
            color: grey;
        }
    }


}

.thumbnail-container:hover {
    opacity: 0.75;
}