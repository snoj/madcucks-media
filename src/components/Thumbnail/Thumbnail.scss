$break-small: 480px;
$break-medium: 720px;

.thumbnail-container {
    
    display: flex;
    flex-direction: column;
    padding-top: 1em;
    padding-bottom: 1em;
    flex: 0 0 32%;

    @media screen and (max-width: $break-medium) {
        flex: 0 0 48%;
    }

    @media screen and (max-width: $break-small) {
        flex: 0 0 94%;
    }

    img {
        flex: 0 0 80%;
    }

    div {
        background-color: black;
        flex: 0 0 20%;
        text-align: left;
        font-size: 0.9em;

        .thumbnail-span-heading {
            color: white;
        }

        .thumbnail-span-subheading {
            color: grey;
        }

    }
}