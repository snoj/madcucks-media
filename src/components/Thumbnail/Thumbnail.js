import React, { Component } from 'react';
import './Thumbnail.css';

export default class Thumbnail extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <div className="thumbnail-container">
                <img src={this.props.imageSrc}/>
                <div>
                    <div className="thumbnail-span-heading">{this.props.title}</div>
                    <div className="thumbnail-span-subheading">Subheading</div>
                </div>
            </div>
        );
    }
}