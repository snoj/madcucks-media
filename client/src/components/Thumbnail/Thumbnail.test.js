import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Thumbnail from './Thumbnail';

const episodeInfo = {
    "title": "Episode 89 - Dick on Getting Caught",
    "published": "2018-02-13T05:09:12.000Z",
    "guid": "04b1b04874583ba659541cc90290fae1",
    "link": "http://dick.show/episode-89/",
    "image": "http://static.libsyn.com/p/assets/a/e/8/9/ae89bf802d4614a7/89.png",
    "description": "<p>People who ignore \"The Zipper\", the mystery of my broken toilet, Smiley my broken handyman, \"My House\" TV has turned me into a dog, Nick Rekieta calls in with some Hot Goss about Maddox's maybe two or three time DUI lawyer \"Dog Bite\" Kevin Landau, Asterios' Cernovich sex tape, a special erotic story for Valentine's Day, Sean learns to ride a bicycle, The Skate Mate, being on The Milo Yiannopoulos show, the feminist caliphate, drunk dogs, and the nature of getting caught; all that and more this week on The Dick Show!</p>",
    "enclosure": {
        "filesize": 133572480,
        "type": "audio/mpeg",
        "url": "http://traffic.libsyn.com/thedickshow/The_Dick_Show_089.mp3?dest-id=516728"
    },
    "duration": "02:18:03"
};

describe('Thumbnail', () => {
    it('renders with episode info', () => {
        const component = shallow(
        <Thumbnail title={episodeInfo.title} 
            duration={episodeInfo.duration} 
            date={episodeInfo.duration}
            imgSrc={episodeInfo.image}
            episodeURL={episodeInfo.link}/>
        );
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    });

    describe('convertDate function', () => {
        it('evaluates to 10 minutes ago', () => {
            const currentDate = new Date("2018-02-13T05:19:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("10 minutes ago");
        });

        it('evaluates to 1 hour ago', () => {
            const currentDate = new Date("2018-02-13T06:09:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("1 hour ago");

        });

        it('evaluates to 4 hours ago', () => {
            const currentDate = new Date("2018-02-13T09:09:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("4 hours ago");

        });

        it('evaluates to 1 day ago', () => {
            const currentDate = new Date("2018-02-14T05:09:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("1 day ago");

        });

        it('evaluates to 5 days ago', () => {
            const currentDate = new Date("2018-02-19T01:09:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("5 days ago");

        });

        it('evaluates to 1 month ago', () => {
            const currentDate = new Date("2018-03-23T01:09:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("1 month ago");

        });

        it('evaluates to 6 months ago', () => {
            const currentDate = new Date("2018-08-13T01:09:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("6 months ago");

        });

        it('evaluates to 1 year ago', () => {
            const currentDate = new Date("2019-02-13T01:09:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("1 year ago");

        });

        it('evaluates to 2 years ago', () => {
            const currentDate = new Date("2020-02-13T01:09:12.000Z");
            const publishedDate = new Date(episodeInfo.published);
            let wrapper = shallow(
                <Thumbnail title={episodeInfo.title} 
                    duration={episodeInfo.duration} 
                    date={episodeInfo.duration}
                    imgSrc={episodeInfo.image}
                    episodeURL={episodeInfo.link}/>
                );
            const result = wrapper.instance().convertDate(currentDate,publishedDate);

            expect(result).toEqual("2 years ago");

        });
    });

});

