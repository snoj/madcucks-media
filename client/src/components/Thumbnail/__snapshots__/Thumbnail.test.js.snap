// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders with episode info 1`] = `
<div
  className="thumbnail-container"
>
  <Link
    replace={false}
    to="http://dick.show/episode-89/"
  >
    <img
      alt=""
    />
  </Link>
  <div
    className="thumbnail-info-container"
  >
    <div
      className="thumbnail-span-heading"
    >
      Episode 89 - Dick on Getting Caught
    </div>
    <div
      className="thumbnail-span-subheading"
    >
      02:18:03
       / 
      NaN years ago
    </div>
  </div>
</div>
`;
